File: C1.UWP.FlexReport/VB/FlexReportExplorer/Data/CustomTemplateSelector.vb
Imports C1.Xaml

Public Class CustomTemplateSelector
    Inherits C1DataTemplateSelector

    Protected Overrides Function SelectTemplateCore(item As Object, container As DependencyObject) As DataTemplate
        Dim group = TryCast(item, Category)
        If group IsNot Nothing Then
            If group.Name = "Separator" Then
                Return TryCast(Resources("CategorySeparatorTemplate"), DataTemplate)
            End If
            Return TryCast(Resources("CategoryTemplate"), DataTemplate)
        End If
        If TypeOf item Is Report Then
            Return TryCast(Resources("ReportTemplate"), DataTemplate)
        End If
        Return Nothing
    End Function

End Class

File: C1.UWP.FlexReport/CS/FlexReportExplorer/Data/CustomTemplateSelector.cs
using C1.Xaml;
using Windows.UI.Xaml;

namespace C1FlexReportExplorer
{
    public class CustomTemplateSelector : C1DataTemplateSelector
    {
        protected override DataTemplate SelectTemplateCore(object item, DependencyObject container)
        {
            var group = item as Category;
            if (group != null)
            {
                if (group.Name == "Separator")
                    return Resources["CategorySeparatorTemplate"] as DataTemplate; 
                return Resources["CategoryTemplate"] as DataTemplate;
            }
            if (item is Report)
            {
                return Resources["ReportTemplate"] as DataTemplate;
            }
            return null;
        }
    }
}
